packages/notebook/src/actions.tsx

```diff
@@ -2292,4 +2410,155 @@ namespace Private {
     }
     cell.value.text = newHeader + source;
   }
+
+  /** Functionality related to collapsible headings */
+  export namespace Headings {
+    /** Find the heading that is parent to cell.
+     *
+     * @param childCell - The cell that is child to the sought heading
+     * @param notebook - The target notebook widget
+     * @param includeChildCell [default=false] - if set to true and childCell is a heading itself, the childCell will be returned
+     * @param returnIdx [default=false] - if set to true, the cell idx is returned rather than the cell object.
+     *
+     * @returns the (index | Cell object) of the parent heading or (-1 | null) if there is no parent heading.
+     */
+    export function findParentHeading(
```

Comment:
@marthacryan is there already some helper functions to get the parent or the sibling header?

> It does not look like it. But I'm not familiar with the heading collapser feature.

Reply:
Actually I think there is. It is the `findNearestParentHeader` (see PR description). But my first impression is that it is buggy so I didn't use it. But I'm not sure if the reaction on markdown cells is intended.